enginetest/queries/join_queries.go

```diff
@@ -19,6 +19,22 @@ import (
 )
 
 var JoinQueryTests = []QueryTest{
```

Comment:
Any other tests we should add? semi and anti merge join? nested subqueries to validate scope handling? more complex/multicolumn indexes?